EmpireLegends/StartingPhase.cpp

```diff
@@ -0,0 +1,326 @@
+		break;
+
+	default: playerCoins = 0;
+	}
+
+	return playerCoins;
+}
+
+string StartingPhase::getRandomContinent()
+{
+	/* initialize random seed: */
+	srand(time(nullptr));
+
+	/* generate secret number between 0 and 2: */
+	int rand = std::rand() % 3;
+
+	string continent = rand == 0 ? "Continent 1" : "Continent 2";
+
+	return continent;
+}
+
+string StartingPhase::getRandomTerritory()
+{
+	string territory = "Region 3";
+	/* initialize random seed: */
+	srand(time(nullptr));
+
+	/* generate secret number between 0 and 2: */
+	int rand = std::rand() % 3;
+
+	if(rand == 0)
+	{
+		territory = "Region 1";
+	}
+	else if(rand == 1)
+	{
+		territory = "Region 2";
+	}
+	
+	return territory;
+}
+
+void StartingPhase::setupCardDeck()
+{
+	// set up cards
+	Card* cardOne = new Card("Dire dragon", new Flying, "Place 3 armies and destroy one army");
+	Card* cardTwo = new Card("Dire giant", new Immune, "Place 3 armies and destroy army");
+	Card* cardThree = new Card("Dire eye", new Flying, "Place 4 armies");
+	Card* cardFour = new Card("Dire Ogre", new CoinVPs, "Move 2 armies");
+	Card* cardFive = new Card("Lake", new SetNameVPs(CardSet::forest), "Place 2 armies and move 3 armies");
+	Card* cardSix = new Card("Noble Hills", new CompleteSetVPs(3, CardSet::noble), "Place 3 armies");
+
+	// add Cards to a vector
+	vector<Card*> cardVector;
+	cardVector.push_back(cardOne);
+	cardVector.push_back(cardTwo);
+	cardVector.push_back(cardThree);
+	cardVector.push_back(cardFour);
+	cardVector.push_back(cardFive);
+	cardVector.push_back(cardSix);
+
+	// Create deck and add cards (this creates a card hand as well)
+	cardDeck = new Deck(cardVector);
+
+	std::cout << *cardDeck;
+
+	cardDeck->shuffle();
+	std::cout << *cardDeck;
+
+	// Draw cards from deck and add them to the hand
+	cardDeck->draw(6);
+	std::cout << *cardDeck;
+}
+
+void StartingPhase::populatePlayers()
+{
+	string username;
+	string color;
+	
+	// determines how many coins per player
+	std::cout << "\nNumber of players in the game (2-4): ";
+	std::cin >> numOfPlayers;
+
+	// Assign number of coins based on players
+	int playerCoins = setNumberOfCoins(numOfPlayers);
+
+	// Get players
+	for (int i = 0; i < 2; i++) {
+		std::cout << "Type player name: ";
+		std::cin >> username;
+
+		// creates a new pointer for each player object and adds it to the player vector
+		players.push_back(new Player(username, playerCoins));
+	}
+
+	// select colors
+	for (int i = 0; i < numOfPlayers; i++)
+	{
+		const Player* player = players[i];
+		Resources* resources = players[i]->getResources();
+
+		std::cout << player->getName() << "Choose your color: ";
+		std::cin >> color;
+
+
+		Color col = colorUtilities->parseColor(color);
+		resources->playerColor = col;
+
+		// mark color as unavailable
+		colorUtilities->setColorAvailability(color, false);
+	}
+}
+
+void StartingPhase::setupStartingTerritories()
+{
+	Territory<Region>* startingRegion = map->findTerritory("Continent 1")->value->findTerritory("Region 1");
```

Comment:
I don't think finding the region by name is correct, since there will be a choice of multiple maps, which can have random names. We should probably select it by index (first territory and first region)